test/desdemona/plugins/swift_test.clj

```diff
@@ -0,0 +1,152 @@
+(ns desdemona.plugins.swift-test
+  (:require [clojure.test :refer [deftest testing is]]
+            [cheshire.core :as json]
+            [camel-snake-kebab.core :refer [->kebab-case-keyword]]
+            [clojure.java.io :as io]
+            [clj-time.core :as t]
+            [clj-http.fake :as http]
+            [onyx.peer.pipeline-extensions :as p-ext]
+            [onyx.peer.function :as function]
+            [desdemona.plugins.swift :as swift])
+  (:import (org.joda.time DateTimeUtils)))
+
+(defmacro freeze-time [time & forms]
+  `(try
+     (DateTimeUtils/setCurrentMillisFixed (.getMillis ~time))
+     ~@forms
+     (finally
+       (DateTimeUtils/setCurrentMillisSystem))))
+
+(deftest get-cloud-files-test
+  (let [cloud-files {:name "cloudFiles" :type "object-store" :id 1}
+        auth-response {:access {:service-catalog [{:name "one"}
+                                                  {:name "two" :type "object-store"}
+                                                  {:name "cloudFiles" :type "something"}
+                                                  cloud-files
+                                                  {:type "object-store"}]}}
+        got (swift/get-cloud-files auth-response)]
+    (is (= cloud-files got))))
+
+(deftest get-token-test
+  (let [token {:id 1}
+        auth-response {:access {:token token}}
+        got (swift/get-token auth-response)]
+    (is (= token got))))
+
+(deftest authenticate-test
+  (testing "This test verifies that the authenticate function makes a request to the
+           URL specified by the auth-url parameter, and includes the username and api-key
+           in the expected JSON format."
+    (http/with-fake-routes
+      {"http://authurl.com" {:post (fn [req]
+                                     (let [body (json/parse-stream (io/reader (req :body)) ->kebab-case-keyword)
+                                           username (-> body :auth :rax-kskey:api-key-credentials :username)
+                                           api-key (-> body :auth :rax-kskey:api-key-credentials :api-key)]
```

Comment:
Nitpick: this is an example of a place you might want to consider a `juxt`; a la:

`[username api-key] (-> ... ((juxt :username :api-key)))`